addons/crm_mail_plugin/controllers/mail_plugin.py

```diff
@@ -53,14 +53,29 @@ def _fetch_partner_leads(self, partner, limit=5, offset=0):
 
     def _get_contact_data(self, partner):
         contact_values = super(MailPluginController, self)._get_contact_data(partner)
+
+        if not request.env.user.has_group('sales_team.group_sale_salesman'):
+            return contact_values
+
         if not partner:
             contact_values['leads'] = []
         else:
             contact_values['leads'] = self._fetch_partner_leads(partner)
         return contact_values
 
     def _mail_content_logging_models_whitelist(self):
-        return super(MailPluginController, self)._mail_content_logging_models_whitelist() + ['crm.lead']
+        models_whitelist = super(MailPluginController, self)._mail_content_logging_models_whitelist()
+        if not request.env.user.has_group('sales_team.group_sale_salesman'):
+            return models_whitelist
+        return models_whitelist + ['crm.lead']
 
     def _translation_modules_whitelist(self):
-        return super(MailPluginController, self)._translation_modules_whitelist() + ['crm_mail_plugin']
+        modules_whitelist = super(MailPluginController, self)._translation_modules_whitelist()
+        if not request.env.user.has_group('sales_team.group_sale_salesman'):
+            return modules_whitelist
+        return modules_whitelist + ['crm_mail_plugin']
+
+    def _user_can_create_contact(self):
```

Comment:
I do not see that method in the v15 controller so I'm confused:
https://github.com/odoo/odoo/blob/15.0/addons%2Fmail_plugin%2Fcontrollers%2Fmail_plugin.py

Maybe I did not look correctly?

Reply:
it should not be there (it was for the second point, but we will do it in a separate task)